vault/resource_consul_secret_backend_role.go

```diff
@@ -19,11 +22,11 @@ var (
 
 func consulSecretBackendRoleResource() *schema.Resource {
 	return &schema.Resource{
-		Create: consulSecretBackendRoleWrite,
-		Read:   ReadWrapper(consulSecretBackendRoleRead),
-		Update: consulSecretBackendRoleWrite,
-		Delete: consulSecretBackendRoleDelete,
-		Exists: consulSecretBackendRoleExists,
+		CreateContext: consulSecretBackendRoleWrite,
+		ReadContext:   ReadContextWrapper(consulSecretBackendRoleRead),
+		UpdateContext: consulSecretBackendRoleWrite,
+		Delete:        consulSecretBackendRoleDelete,
+		Exists:        consulSecretBackendRoleExists,
```

Comment:
These were updated to Context functions to provide a context to the semver lookup. I opted to only update the ones affected by the need for this code change, but I could see the argument for updating `Delete` and `Exists` as well, to not leave any un-updated portions.

Reply:
Please go ahead and update all of them.

Reply:
Do the underlying logical operations needs to be updated to use the Context version?

For example on line 205 in write:
`client.Logical().Write(path, data)`
to
`client.Logical().WriteWithContext(ctx, path, data)`

Reply:
Shouldn't be necessary.

Reply:
That's correct. `Exists` is deprecated and may be removed in favor of the Context functions :)